Loop/Window Management/Window/WindowUtility+FocusNavigation.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  WindowUtility+FocusNavigation.swift
+//  Loop
+//
+//  Created by cipher-shad0w on 2025-10-30.
+//
+
+import AppKit
+import OSLog
+import SwiftUI
+
+extension WindowUtility {
+    /// Finds the next window to focus in the specified direction.
+    /// - Parameters:
+    ///   - currentWindow: The currently focused window to navigate from, or nil to navigate from screen center
+    ///   - edge: The direction to search for the next window (leading, trailing, top, bottom)
+    /// - Returns: The next window in the specified direction, or `nil` if no suitable window is found
+    static func directionalWindow(from currentWindow: Window?, edge: Edge) -> Window? {
+        let allWindows = windowList()
+
+        // If no current window, navigate from screen center
+        if currentWindow == nil {
+            guard let screen = NSScreen.screenWithMouse ?? NSScreen.main else {
+                logger.error("[FocusNavigation] Could not determine active screen")
+                return nil
+            }
+
+            let screenCenter = screen.frame.center
+            logger.info("[FocusNavigation] Navigating from screen center: (\(screenCenter.x), \(screenCenter.y))")
+
+            // Filter to get only visible, non-minimized, non-excluded windows
+            let availableWindows = allWindows.filter { window in
+                !window.minimized &&
+                    !window.isWindowHidden &&
+                    !window.isAppExcluded
+            }
+
+            guard !availableWindows.isEmpty else {
+                logger.info("[FocusNavigation] No windows available to focus")
+                return nil
+            }
+
+            // Find the closest window in the specified direction from screen center
+            let nextWindow = availableWindows
+                .filter { window in
+                    isInDirection(window.frame, from: screenCenter, edge: edge)
+                }
+                .min { window1, window2 in
+                    screenCenter.distance(to: window1.frame.center) < screenCenter.distance(to: window2.frame.center)
+                }
+
+            if let nextWindow {
+                let nextWindowName = nextWindow.nsRunningApplication?.localizedName ?? nextWindow.title ?? "<unknown>"
+                logger.info("[FocusNavigation] Found window to focus in direction \(String(describing: edge)): \(nextWindowName)")
```

Comment:
I know the rest of the codebase hasn't been fully updated to use this, but prefer:
```suggestion
                logger.info("[FocusNavigation] Found window to focus in direction \(String(describing: edge)): \(nextWindow.debugDescription)")
```